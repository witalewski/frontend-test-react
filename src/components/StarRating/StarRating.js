import React from 'react';

const StarRating = ({ rating }) => {
    const stars = [];
    for (let i = 1; i<= 5; i++) {
        let className;
        if (i >= rating + 1) {
            className = "star--empty";
        } else if (i > rating ) {
            className = "star--half";
        } 
        stars.push(<use xlinkHref="#icon-star" key={`star-${i}-${className}`} x={(i-1)*36} y="0" className={className || ''} />)
    }
  return (
    <div>
      <div className="star-rating">
        <p>
          <svg class="star" viewBox="0 0 180 32">
            {stars}
          </svg>
        </p>
      </div>
      ({rating})
    </div>
  );
};

export { StarRating };
