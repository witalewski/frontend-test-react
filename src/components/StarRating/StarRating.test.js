import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MOCK_RESTAURANT from '../../fixtures/restaurant.json';
import { StarRating } from './StarRating';

describe('StarRating', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <StarRating rating={MOCK_RESTAURANT.rating} />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
