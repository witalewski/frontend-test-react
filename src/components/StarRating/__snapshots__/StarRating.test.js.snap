// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating matches snapshot 1`] = `
<div
  className="star-rating"
>
  <svg
    className="star"
    viewBox="0 0 180 32"
  >
    <use
      className=""
      x={0}
      xlinkHref="#icon-star"
      y="0"
    />
    <use
      className=""
      x={36}
      xlinkHref="#icon-star"
      y="0"
    />
    <use
      className=""
      x={72}
      xlinkHref="#icon-star"
      y="0"
    />
    <use
      className=""
      x={108}
      xlinkHref="#icon-star"
      y="0"
    />
    <use
      className="star--half"
      x={144}
      xlinkHref="#icon-star"
      y="0"
    />
  </svg>
</div>
`;
